python_modules/dagster/dagster/_grpc/server.py

```diff
@@ -1115,6 +1115,13 @@ def open_server_process(
     if loadable_target_origin:
         subprocess_args += loadable_target_origin.get_cli_args()
 
+    # Unset click environment variables in the current environment
+    # that might conflict with arguments that we're using
+    env = {
+        **(env or {}),
+        **({"DAGSTER_GRPC_SOCKET": ""} if port else {"DAGSTER_GRPC_PORT": ""}),
```

Comment:
did you mean `**({"DAGSTER_GRPC_SOCKET": port} if port else {"DAGSTER_GRPC_PORT": ""}),`

Reply:
The goal is to empty out the socket env var in the subprocess if port is being explicitly set by the parent process (and vice versa)

Reply:
should we delete the env vars instead of setting them to ""? its a little odd to be inserting empty str env vars. Also maybe warn / print something if they are getting removed?